Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverInformation.hs

```diff
@@ -48,6 +48,23 @@ getEnabledAt driverId = do
   dInfo <- findById driverId
   return (dInfo >>= (.enabledAt))
 
+findAllByEnabledAtInWindow :: MonadFlow m => Id Merchant -> Maybe UTCTime -> Maybe UTCTime -> m [DriverInformation]
+findAllByEnabledAtInWindow merchantId from to = do
+  findAllWithKV
+    [ Se.Or
+        [ Se.And
+            [ Se.Is BeamDI.enabledAt $ Se.GreaterThanOrEq from,
+              Se.Is BeamDI.enabledAt $ Se.LessThanOrEq to,
+              Se.Is BeamDI.merchantId $ Se.Eq (Just merchantId.getId)
+            ],
+          Se.And
+            [ Se.Is BeamDI.lastEnabledOn $ Se.GreaterThanOrEq from,
```

Comment:
Why `lastEnabled` check? Is it because of the backfilling we did for this column?

Reply:
for calculating the freeTrialDays left in driverInfoResponse, we are using both the columns and if both columns are null then only now will be passed. So same logic is used for query.

Reply:
<img width="1581" alt="Screenshot 2023-10-06 at 7 02 00 PM" src="https://github.com/nammayatri/nammayatri/assets/33424495/ca22fa8b-7e0c-401b-93d4-b2514316af66">

Reply:
The query is for OR.
Either - enabledAt in the startTime endTime range
Or - lastEnabledOn in the startTime endTime range

Reply:
We are interested in the first time the driver was enabled. And we have values for lastEnabled whenever we have enabled field populated. I still don't see any reason to use it

Reply:
Okay I will update the query.